packages/slate-react/src/components/content.js

```diff
@@ -161,20 +162,33 @@ class Content extends React.Component {
     }
 
     // If the new range matches the current selection, do nothing.
-    if (
-      current &&
-      range.startContainer == current.startContainer &&
-      range.startOffset == current.startOffset &&
-      range.endContainer == current.endContainer &&
-      range.endOffset == current.endOffset
-    ) {
-      return
+    if (current) {
+      if (
+        range.startContainer == current.startContainer &&
+        range.startOffset == current.startOffset &&
+        range.endContainer == current.endContainer &&
+        range.endOffset == current.endOffset
+      ) {
+        return
+      }
+      if (
+        range.startContainer == current.endContainer &&
+        range.startOffset == current.endOffset &&
+        range.endContainer == current.startContainer &&
+        range.endOffset == current.startOffset
+      ) {
+        return
+      }
     }
 
     // Otherwise, set the `isUpdatingSelection` flag and update the selection.
     this.tmp.isUpdatingSelection = true
     native.removeAllRanges()
-    native.addRange(range)
+    if (isBackward) {
+      native.setBaseAndExtent(range.endContainer, range.endOffset, range.startContainer, range.startOffset)
+    } else {
+      native.setBaseAndExtent(range.startContainer, range.startOffset, range.endContainer, range.endOffset)
+    }
```

Comment:
And then if the util was fixed, wouldn't this code be unnecessary?

Reply:
same explanation, the range object is always a forward range. seems to be Range API implementation issue